### PR TITLE
Share modal refactor

### PR DESCRIPTION
**Pull request by Hyperkid123**
jira: https://projects.engineering.redhat.com/browse/SSP-1681

### Changes
- the group list is now dynamic
  - can add multiple groups at once
  - user must use the `+` button to add the group
- new icon for group deletion
- submit button is disabled until the form is in `dirty` state
 

share

### CODE REVIEW

File: src/forms/form-fields/share-group-select.js [hunk 1]
```diff
@@ -1,77 +1,127 @@
...
+              isDisabled={!group || !permission}
+              onClick={() => {
+                addGroup({
+                  groupName: group.label,
+                  group_uuid: group.value,
+                  permissions: permission
+                });
+                setPermission(undefined);
+                setGroup(undefined);
+                setResetGroup((prev) => prev + 1);
```
Comment (rvsia) [inline src/forms/form-fields/share-group-select.js:100]: use `useReducer` instead? :thinking:

**Comment by Hyperkid123** [comment]
I will squash the messages as well.

**Comment by Hyperkid123** [comment]
I've updated the sharing success message. We could add the numbers for share/unshare but honestly, I feel like this message is far better, considering that when user only updates the actually shares/unshares the portfolio which is very confusing.

screenshot-ci foo redhat com_1337-2020 07 27-10_00_54

**Comment by gmcculloug** [comment]
I like the single message but want to confirm with @sbuenafe-rh.  Also, what does the failure message look like?

**Comment by Hyperkid123** [comment]
The failure message is a problem. Since we have to create one API call per each share/unshare the user will get an error message only for the API call that failed. So you will get whatever error message the API returns.

In addition, if one of the calls fails, the rest of share/unshare actions will be persisted. Obviously we can't rollback the database form UI at that point.

**Comment by lgalis** [comment]
@Hyperkid123 - I will merge this PR, as it works well - and the issue in my comment above will be fixed in #682

**Pull request merged**